src/Forms/GridField/GridFieldAddNewButton.php

```diff
@@ -36,7 +37,16 @@ public function __construct($targetFragment = 'before')
 
     public function getHTMLFragments($gridField)
     {
-        $singleton = singleton($gridField->getModelClass());
+        $modelClass = $gridField->getModelClass();
+        $singleton = singleton($modelClass);
+
+        if (!$singleton->hasMethod('canCreate')) {
+            throw new LogicException(
+                __CLASS__ . ' cannot be used with models that do not implement canCreate().'
+                . " Remove this component from your GridField or implement canCreate() on $modelClass"
+            );
+        }
```

Comment:
This and other similar exceptions don't break BC. An exception would already have been thrown, because it was trying to call a method that doesn't exist. This just provides more explicit messaging for the developer to know how to resolve the problem.